test/contexts_spec.lua

```diff
@@ -114,6 +114,49 @@ for _, lang in ipairs(langs_with_queries) do
       setup(function()
         cmd([[let $XDG_CACHE_HOME='scratch/cache']])
         install_langs(lang)
+        cmd('edit ' .. test_file)
+        local bufnr = api.nvim_get_current_buf()
+        exec_lua(
```

Comment:
Can you add some comments around this. At a glance, I have no idea what this code is trying to do.

Also, you should be able to replace the string in `exec_lua` with a function.

Reply:
I replaced with a function, a decent chunk stayed in `exec_lua` but definitely easier to read